### PR TITLE
Change CreateFileTemplateService to take document moniker

### PR DESCRIPTION
**Pull request by davkean**
In preparation for retrieving app designer folder from a special file handler (in a future change), which returns a path, we now take a "parentDocumentMoniker" instead of a IProjectTree to CreateFileFromTemplateService to avoid having to look up in the tree multiple times.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/CreateFileFromTemplateService.cs [hunk 1]
```diff
@@ -34,13 +34,13 @@ public CreateFileFromTemplateService(IUnconfiguredProjectVsServices projectVsSer
...
-        /// <param name="parentNode">The node to which the new file will be added.</param>
+        /// <param name="parentDocumentMoniker">The path to the node to which the new file will be added.</param>
         /// <param name="fileName">The name of the file to be created.</param>
         /// <returns>true if file is added successfully.</returns>
-        public async Task<bool> CreateFileAsync(string templateFile, IProjectTree parentNode, string fileName)
+        public async Task<bool> CreateFileAsync(string templateFile, string parentDocumentMoniker, string fileName)
         {
             Requires.NotNull(templateFile, nameof(templateFile));
-            Requires.NotNull(parentNode, nameof(parentNode));
+            Requires.NotNullOrEmpty(parentDocumentMoniker, nameof(parentDocumentMoniker));
```
Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/CreateFileFromTemplateService.cs]: Should be `NotNullOrWhitespace`.

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/CreateFileFromTemplateService.cs]: What's the relationship between the unicode standard for whitespace (which is what NotNullOrWhitespace enforces) and paths? This isn't for enforcing "good paths" - this is for finding bugs (you've passed me a path that isn't set "empty and null are interchangeable in .NET apis"). Bad paths will be caught later by other .NET/Win32 APIs .

Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/CreateFileFromTemplateService.cs]: Here's a little sample that shows you what I mean:

``` C#
    class Program
    {
        static void Main(string[] args)
        {
            for (char c = (char)0; c < Char.MaxValue; c++)
            {
                if (char.IsWhiteSpace(c) && CanCreateFile(c))
                {
                    Console.WriteLine("Valid Path Char: " + ToCodePoint(c));
                }
            }
        }

        private static string ToCodePoint(char c)
        {
            return string.Format("\\u{0:x4} ", (int)c);
        }

        private static bool CanCreateFile(char c)
        {
            try
            {
                string path = Path.Combine(@"c:\Temp", c.ToString());

                File.WriteAllText(path, "");
            }
            catch (DirectoryNotFoundException)
            {   // Handle ' ' and '\t', etc 
                return false;
            }
            catch (ArgumentException)
            {
                return false;
            }

            return true;
        }
    }
```

This outputs the following:

```
Valid Path Char: \u1680
Valid Path Char: \u2000
Valid Path Char: \u2001
Valid Path Char: \u2002
Valid Path Char: \u2003
Valid Path Char: \u2004
Valid Path Char: \u2005
Valid Path Char: \u2006
Valid Path Char: \u2007
Valid Path Char: \u2008
Valid Path Char: \u2009
Valid Path Char: \u200a
Valid Path Char: \u2028
Valid Path Char: \u2029
Valid Path Char: \u202f
Valid Path Char: \u205f
Valid Path Char: \u3000
```

These "whitespace" chars are all valid file name chars on my NTFS drive, yet IsNullOrWhiteSpace would rule them out.

**Pull request merged**